modules/functions.js

```diff
@@ -0,0 +1,260 @@
+
+  })
+
+  if (!priceOfTv) {
+
+    const msgNoPrice = "No price found" 
+    return msgNoPrice;
+
+   }
+
+  return `€${priceOfTv.price},-`;
+
+}
+
+  // ------------------------- Opdracht 4c ----------------------
+
+function screenSizeTv (arr, name) {
+
+  const sizeOfTv = arr.find((size) => {
+
+    return size.brand.toLowerCase() === name;
+    
+  })
+
+  if (!sizeOfTv) {
+
+    const msgNoSize = "No screensize found" 
+    return msgNoSize;
+   }
+
+   const sizeNewArray = sizeOfTv.availableSizes;
+   let availableSize = '';
+
+   for (let i = 0; i < sizeNewArray.length; i++) {
+
+      const inchCm = sizeNewArray[i] * 2.54;
+      // samenvoegen string met nieuwe data omzetting naar inch en afronden
+      availableSize +=  `${sizeNewArray[i]} inch (${Math.round(inchCm)} cm)`;
+      //  Doordat we niet kunnen eindigen met een | zullen we een if statement gebruiken waarin de index i kleiner is dan de array. Per item wordt | toegevoegd en door -1 de laatste niet. Op deze manier kunnen we een karakter tussen de tekst in plaatsen en de laatste negeren.
+      if (i < sizeNewArray.length -1) {
+        availableSize += ' | '
+      }
+
+      // console.log(sizeNewArray[i])
+   }
+
+//  console.log(availableSize);
+  
+  return availableSize;
+}
+
+
+ // ------------------------- Opdracht 4d ----------------------
+
+  function TvSpecification (arr, name) {
+
+    const specsOfTv = arr.find((specs) => {
+  
+      return specs.brand.toLowerCase() === name;
+      
+    })
+  
+    if (!specsOfTv) {
+  
+      const msgNoSpecs = "No Television with this brand found" 
+      return msgNoSpecs;
+     }
+  
+     const specsNewArray = specsOfTv.availableSizes;
+     let availableScreenSize = '';
+  
+     for (let i = 0; i < specsNewArray.length; i++) {
+  
+        const inchCm = specsNewArray[i] * 2.54;
+        // samenvoegen string met nieuwe data omzetting naar inch en afronden
+        availableScreenSize +=  `${specsNewArray[i]} inch (${Math.round(inchCm)} cm)`;
+        //  Doordat we niet kunnen eindigen met een | zullen we een if statement gebruiken waarin de index i kleiner is dan de array. Per item wordt | toegevoegd en door -1 de laatste niet. Op deze manier kunnen we een karakter tussen de tekst in plaatsen en de laatste negeren.
+        if (i < specsNewArray.length -1) {
+          availableScreenSize += ' | '
+        }
+  
+        // console.log(sizeNewArray[i])
+     }
+  
+  //  console.log(availableSize);
+    
+    return `
+   -------------------------------------------------------
+
+   ${specsOfTv.brand} ${specsOfTv.type} - ${specsOfTv.name}
+   €${specsOfTv.price},-
+   ${availableScreenSize};
+  
+   -------------------------------------------------------
+    `
+    
+  }
+
+  
+ // ------------------------- Opdracht 4e ----------------------
+
+
+  function TvSpecAll (arr) {
+    // Array map methode om informatie uit de array te halen
+    const specsAllTv = arr.map((specs) => {
+       return specs;
+    })
```

Comment:
Hier maak je feitelijk een clone van `arr`. Was dat ook je intentie? Zo niet, dan had je ook kunnen werken met `arr` in plaats van `specsAllTv`

Reply:
Ik vond het mooier om arr in een constante te bewaren en deze dan aan te roepen. Ik zie ook in dat dit korter geschreven kan worden. Dus wel met een intentie gedaan.